app/services/location_comparer.rb

```diff
@@ -0,0 +1,31 @@
+class LocationComparer
```

Comment:
I am not sure this class is necessary right now.  Instead of this, you could do:
```ruby
old_address = person.location
new_address = Location.new(zip_code: zip) #etc
old_address.attrbutes.except(:person_id) == new_address.attributes.except(:person_id)
```
This is simplified, but you get the idea.  

At a future point, possibly very soon, we'll need access to more advance business logic.  Maybe we should just plan for that now??   Anyway, at that point, I'd have a separate Class/Objects for comparing.  However, instead of a class instantiating single instances to do the comparing, I'd recommend instantiating multiple objects of the same class and have compare those objects.

For example, you could have `LocationComparable` and do.
```ruby
class LocationComparable
  attr_accessor :city, :state, :zip

  def initialize(city:,state:,zip:)
    #blabla
  end

  different_from?(other_object)
    self.instance_values == other_object.instance_values
  end
end

old_location = LocationComparable.new(attrs)
new_location = LocationComparable.new(attrs)
old_location.different_from?(new_location)
```

We could even make this private if there was a resonable place to do so.  https://blog.arkency.com/2016/02/private-classes-in-ruby/